app/services/accounting/accept_service.rb

```diff
@@ -0,0 +1,94 @@
+module Accounting
+  class AcceptService
+    include AuthorizeNet::API
+    include AccountingHelper
+
+    attr_accessor :params
+    attr_accessor :profile, :payment
+
+    def initialize(params, profile)
+      @params = params
+      @profile = profile
+    end
+
+    ##
+    # 1. Validate Payment Address before Authorize.net request
+    # 2. Create Authorize.net request to create payment profile
+    #
+    # @author Ming <ming@commercekitchen.com>
+    def build_payment
+      @payment = @profile.payments.build(accept: true, profile_type: @params[:profile_type])
+      if @payment.card?
+        @payment.assign_attributes(@params[:card])
+        return false if @payment.address && @payment.address.invalid?
+      end
+
+      create_customer_payment_profile
+    end
+
+    def save
+      build_payment
+      @payment.errors.blank? && @payment.save
+    end
+
+    def errors
+      @payment.errors
+    end
+
+    private
+
+      ##
+      # Create Authorize.net createCustomerPaymentProfile Xml request
+      # and get newly created payment profile
+      # 
+      # @author Ming <ming@commercekitchen.com>
+      #
+      # @reference https://github.com/AuthorizeNet/sample-code-ruby/blob/master/CustomerProfiles/create-customer-payment-profile.rb
+      def create_customer_payment_profile
+        transaction = Accounting.api(:api, api_options(@profile.accountable))
+        @response = transaction.create_customer_payment_profile(build_request)
+
+        if @response != nil
+          if @response.messages.resultCode == MessageTypeEnum::Ok
+            Accounting.log 'Payment', 'Accept', info: "Successfully created a customer payment profile with id: #{@response.customerPaymentProfileId}."
+            @payment.assign_attributes(
+              title: parsed_response[:card_type],
+              payment_profile_id: @response.customerPaymentProfileId,
+              default: @profile.payments.count == 0,
+              last_four: parsed_response[:account_number].to_s[-4..-1]
```

Comment:
This makes me think we are getting the full card number at this point, is that true?

Reply:
No, API returns masked card number like: xxxxx1234.